frontend/src/pages/RequestDetailsPage/v2/BodyViewerV2.tsx

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from "react";
+import { TextOrJsonViewer } from "../TextJsonViewer";
+
+export function BodyViewerV2({
+  body,
+  contentType,
+  collapsed,
+}: { body: string; contentType?: string; collapsed?: boolean }) {
+  if (contentType?.includes("multipart/form-data")) {
+    return <FormDataViewer body={body} />;
+  }
+
+  return <TextOrJsonViewer text={body} collapsed={collapsed} />;
+}
+
+function FormDataViewer({ body }: { body: string }) {
+  const formData = useMemo(() => {
+    try {
+      const parsedBody = JSON.parse(body);
+      if (parsedBody && typeof parsedBody === "object") {
+        const formDataString = Object.entries(parsedBody)
+          .map(([key, value]) => {
+            if (typeof value === "object" && value !== null) {
+              // Handle nested objects or arrays
+              return `${key}=${JSON.stringify(value)}`;
```

Comment:
probably an edge case, but this will break on a circular JSON object

Reply:
ah yes but this is dealing with something that's already been json serialized, but i will put a note